plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/data/PostgreDateTimeValueHandler.java

```diff
@@ -64,15 +66,18 @@ public Object getValueFromObject(@NotNull DBCSession session, @NotNull DBSTypedO
 
     @Override
     public void bindValueObject(@NotNull DBCSession session, @NotNull DBCStatement statement, @NotNull DBSTypedObject type, int index, Object value) throws DBCException {
-        if (value instanceof String) {
-            try {
-                ((JDBCPreparedStatement)statement).setObject(index + 1, value.toString(), Types.OTHER);
-            }
-            catch (SQLException e) {
-                throw new DBCException(ModelMessages.model_jdbc_exception_could_not_bind_statement_parameter, e);
-            }
-            return;
-        }
+	if (value instanceof String) {
+	    try {
+		if (((PostgreDataSource) session.getDataSource()).getServerType() instanceof PostgreServerRedshift) {
+		    ((JDBCPreparedStatement) statement).setObject(index + 1, value.toString(), Types.VARCHAR);
+		} else {
+		    ((JDBCPreparedStatement) statement).setObject(index + 1, value.toString(), Types.OTHER);
+		}
+	    } catch (SQLException e) {
+		throw new DBCException(ModelMessages.model_jdbc_exception_could_not_bind_statement_parameter, e);
+	    }
+	    return;
+	}
```

Comment:
Please reformat indentation using spaces instead of tabs and then squash these commits into just one